engine/src/main/java/io/zeebe/engine/processing/bpmn/container/SubProcessProcessor.java

```diff
@@ -116,31 +116,46 @@ public void onChildTerminated(
       final ExecutableFlowElementContainer element,
       final BpmnElementContext subProcessContext,
       final BpmnElementContext childContext) {
-
-    eventSubscriptionBehavior
-        .findEventTrigger(subProcessContext)
-        .ifPresentOrElse(
-            eventTrigger -> {
-              if (subProcessContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING
-                  // child context can be null if no child's exist on termination
-                  && (childContext == null || stateBehavior.canBeTerminated(childContext))) {
-                final var terminated =
-                    stateTransitionBehavior.transitionToTerminated(subProcessContext);
-                eventSubscriptionBehavior.activateTriggeredEvent(
-                    subProcessContext.getFlowScopeKey(), terminated, eventTrigger);
-              } else {
-                eventSubscriptionBehavior.activateTriggeredEvent(
-                    subProcessContext.getElementInstanceKey(), subProcessContext, eventTrigger);
-              }
-            },
-            () -> {
-              if (subProcessContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING
-                  // child context can be null if no child's exist on termination
-                  && (childContext == null || stateBehavior.canBeTerminated(childContext))) {
-                final var terminated =
-                    stateTransitionBehavior.transitionToTerminated(subProcessContext);
-                stateTransitionBehavior.onElementTerminated(element, terminated);
-              }
-            });
+    if (subProcessContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING) {
+
+      if (childContext == null || stateBehavior.canBeTerminated(childContext)) {
+        // if we are able to terminate we try to trigger boundary events
+        eventSubscriptionBehavior
+            .findEventTrigger(subProcessContext)
+            .filter(
+                eventTrigger ->
+                    element.getBoundaryEvents().stream()
+                        .anyMatch(b -> b.getId().equals(eventTrigger.getElementId())))
```

Comment:
Do we still need this check? The tests are green without it.

I guess that the bug is fixed by removing the previous event trigger.